Build fails after applying recommended `android_gradle_plugin` upgrade from `"8.2.2"` to `"8.3.0"`

```
Executing tasks: [:app:assembleDebug, :app:assembleDebugUnitTest, :app:assembleDebugAndroidTest, :features:tracker:assembleDebug, :features:tracker:assembleDebugUnitTest, :features:tracker:assembleDebugAndroidTest, :domain:assembleDebug, :domain:assembleDebugUnitTest, :domain:assembleDebugAndroidTest, :resources:assembleDebug, :resources:assembleDebugUnitTest, :resources:assembleDebugAndroidTest, :shared:assembleDebug, :shared:assembleDebugUnitTest, :shared:assembleDebugAndroidTest, :data:datastore:assembleDebug, :data:datastore:assembleDebugUnitTest, :data:datastore:assembleDebugAndroidTest, :data:local:assembleDebug, :data:local:assembleDebugUnitTest, :data:local:assembleDebugAndroidTest, :data:repository:assembleDebug, :data:repository:assembleDebugUnitTest, :data:repository:assembleDebugAndroidTest, :features:alarm:assembleDebug, :features:alarm:assembleDebugUnitTest, :features:alarm:assembleDebugAndroidTest, :features:alarm-api:assembleDebug, :features:alarm-api:assembleDebugUnitTest, :features:alarm-api:assembleDebugAndroidTest, :features:category:assembleDebug, :features:category:assembleDebugUnitTest, :features:category:assembleDebugAndroidTest, :features:category-api:assembleDebug, :features:category-api:assembleDebugUnitTest, :features:category-api:assembleDebugAndroidTest, :features:glance:assembleDebug, :features:glance:assembleDebugUnitTest, :features:glance:assembleDebugAndroidTest, :features:home:assembleDebug, :features:home:assembleDebugUnitTest, :features:home:assembleDebugAndroidTest, :features:preference:assembleDebug, :features:preference:assembleDebugUnitTest, :features:preference:assembleDebugAndroidTest, :features:search:assembleDebug, :features:search:assembleDebugUnitTest, :features:search:assembleDebugAndroidTest, :features:task:assembleDebug, :features:task:assembleDebugUnitTest, :features:task:assembleDebugAndroidTest, :libraries:android-test:assembleDebug, :libraries:android-test:assembleDebugUnitTest, :libraries:android-test:assembleDebugAndroidTest, :libraries:appstate:assembleDebug, :libraries:appstate:assembleDebugUnitTest, :libraries:appstate:assembleDebugAndroidTest, :libraries:coroutines:assembleDebug, :libraries:coroutines:assembleDebugUnitTest, :libraries:coroutines:assembleDebugAndroidTest, :libraries:designsystem:assembleDebug, :libraries:designsystem:assembleDebugUnitTest, :libraries:designsystem:assembleDebugAndroidTest, :libraries:di:assembleDebug, :libraries:di:assembleDebugUnitTest, :libraries:di:assembleDebugAndroidTest, :libraries:navigation:assembleDebug, :libraries:navigation:assembleDebugUnitTest, :libraries:navigation:assembleDebugAndroidTest, :libraries:splitInstall:assembleDebug, :libraries:splitInstall:assembleDebugUnitTest, :libraries:splitInstall:assembleDebugAndroidTest, :libraries:test:assembleDebug, :libraries:test:assembleDebugUnitTest, :libraries:test:assembleDebugAndroidTest, :plugins:assemble, :plugins:testClasses, :domain:assemble, :domain:testClasses, :resources:assemble, :resources:testClasses, :shared:assemble, :shared:testClasses, :data:datastore:assemble, :data:datastore:testClasses, :data:local:assemble, :data:local:testClasses, :data:repository:assemble, :data:repository:testClasses, :features:alarm:assemble, :features:alarm:testClasses, :features:alarm-api:assemble, :features:alarm-api:testClasses, :features:category:assemble, :features:category:testClasses, :features:category-api:assemble, :features:category-api:testClasses, :features:home:assemble, :features:home:testClasses, :features:preference:assemble, :features:preference:testClasses, :features:search:assemble, :features:search:testClasses, :features:task:assemble, :features:task:testClasses, :features:tracker:assemble, :features:tracker:testClasses, :libraries:appstate:assemble, :libraries:appstate:testClasses, :libraries:coroutines:assemble, :libraries:coroutines:testClasses, :libraries:designsystem:assemble, :libraries:designsystem:testClasses, :libraries:di:assemble, :libraries:di:testClasses, :libraries:navigation:assemble, :libraries:navigation:testClasses, :libraries:test:assemble, :libraries:test:testClasses] in project /alkaa

Type-safe project accessors is an incubating feature.
Project accessors enabled, but root project name not explicitly set for 'alkaa'. Checking out the project in different folders will impact the generated code and implicitly the buildscript classpath, breaking caching.

> Configure project :
Project accessors enabled, but root project name not explicitly set for 'plugins'. Checking out the project in different folders will impact the generated code and implicitly the buildscript classpath, breaking caching.
The Dependency Analysis plugin is only known to work with versions of AGP between 7.4.2 and 8.2.0-alpha16. You are using 8.3.0. Proceed at your own risk.

> Task :plugins:generateExternalPluginSpecBuilders UP-TO-DATE
> Task :plugins:extractPrecompiledScriptPluginPlugins UP-TO-DATE
> Task :plugins:compilePluginsBlocks UP-TO-DATE
> Task :plugins:generatePrecompiledScriptPluginAccessors UP-TO-DATE
> Task :plugins:generateScriptPluginAdapters UP-TO-DATE
> Task :plugins:pluginDescriptors UP-TO-DATE
> Task :plugins:processResources UP-TO-DATE

> Task :plugins:compileKotlin FAILED
e: file:///alkaa/plugins/src/main/java/extension/CommonExtension.kt:7:20 6 type arguments expected for interface CommonExtension<BuildFeaturesT : BuildFeatures, BuildTypeT : BuildType, DefaultConfigT : DefaultConfig, ProductFlavorT : ProductFlavor, AndroidResourcesT : AndroidResources, InstallationT : Installation>
e: file:///alkaa/plugins/src/main/java/extension/CommonExtension.kt:10:9 Variable expected
e: file:///alkaa/plugins/src/main/java/extension/CommonExtension.kt:11:9 Variable expected
e: file:///alkaa/plugins/src/main/java/extension/CommonExtension.kt:12:9 Variable expected
e: file:///alkaa/plugins/src/main/java/extension/CommonExtension.kt:16:19 Variable expected
e: file:///alkaa/plugins/src/main/java/extension/CommonExtension.kt:20:20 6 type arguments expected for interface CommonExtension<BuildFeaturesT : BuildFeatures, BuildTypeT : BuildType, DefaultConfigT : DefaultConfig, ProductFlavorT : ProductFlavor, AndroidResourcesT : AndroidResources, InstallationT : Installation>
e: file:///alkaa/plugins/src/main/java/extension/CommonExtension.kt:22:9 Variable expected
e: file:///alkaa/plugins/src/main/java/extension/CommonExtension.kt:26:9 Variable expected

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':plugins:compileKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 7s
8 actionable tasks: 1 executed, 7 up-to-date
```
Running Android Studio (Iguana - 2023.2.1) on Ubuntu 23.10

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -2,7 +2,7 @@
 # Project
 version_code = "30000"
 version_name = "3.0.0"
-android_gradle_plugin = "8.2.1"
+android_gradle_plugin = "8.3.2"
 kotlin = "1.9.20"
 android_sdk_compile = "34"
 android_sdk_target = "33"

--- a/plugins/src/main/java/extension/CommonExtension.kt
+++ b/plugins/src/main/java/extension/CommonExtension.kt
@@ -4,7 +4,7 @@ import com.android.build.api.dsl.CommonExtension
 import com.android.build.api.dsl.LibraryExtension
 import org.gradle.api.artifacts.VersionCatalog
 
-fun CommonExtension<*, *, *, *, *>.androidConfig(libs: VersionCatalog) {
+fun CommonExtension<*, *, *, *, *, *>.androidConfig(libs: VersionCatalog) {
 
     defaultConfig {
         compileSdk = Integer.parseInt(libs.sdkCompile)
@@ -17,7 +17,7 @@ fun CommonExtension<*, *, *, *, *>.androidConfig(libs: VersionCatalog) {
     }
 }
 
-fun CommonExtension<*, *, *, *, *>.composeConfig(libs: VersionCatalog) {
+fun CommonExtension<*, *, *, *, *, *>.composeConfig(libs: VersionCatalog) {
     buildFeatures {
         compose = true
     }

